rascal-lsp/src/main/rascal/lang/rascal/lsp/refactor/WorkspaceInfo.rsc

```diff
@@ -294,7 +293,7 @@ set[loc] rascalGetOverloadedDefs(WorkspaceInfo ws, set[loc] defs, MayOverloadFun
             defPaths = defPathStep o selectedFields;
         } else {
             // Find definitions in the reached scope, and definitions within those definitions (transitively)
-            defPaths = defPathStep o invert(defScopes);
+            defPaths = defPathStep o (ws.defines<idRole, scope, defined>)[role]+;
```

Comment:
This is the main performance improvement; way less potential overloads to check.